_posts/2023-07-07-announcing-ballerina-2201.7.0-swan-lake-update-7.md

```diff
@@ -0,0 +1,121 @@
+title: Announcing Ballerina 2201.7.0 (Swan Lake Update 7)
+author: Ballerina Team
+published-date: 7 July 2023
+status: Published
+socialmediaimage: Ballerina-Swan-Lake-GA-Release-banner-02-with-button.png
+permalink: /posts/2023-07-07-announcing-ballerina-2201.7.0-swan-lake-update-7/
+---
+
+<style>.cBlogContent p{white-space: break-spaces !important;}</style>
+
+We are excited to announce the [Ballerina 2201.7.0 (Swan Lake Update 7)](https://ballerina.io/downloads/) release, which adds many new features and improvements to the Ballerina programming language. 
+
+The two main highlights of this release are the support for generating GraalVM native executables and adding aggregation and grouping capabilities to Ballerina query expressions and query actions, which are described below.
+
+## Generation of GraalVM native executables
+
+The Swan Lake Update 7 release comes with official support for generating GraalVM native executables for Ballerina. This new feature allows you to compile Ballerina programs into standalone native executables.
+
+By leveraging GraalVM native images, Ballerina programs can achieve performance improvements and reduced startup times compared to traditional Java Virtual Machine(JVM) execution. This is particularly beneficial for applications with strict latency requirements or resource-constrained environments.
+You can generate a GraalVM native executable for your Ballerina project by executing the command below.
+
+```
+$ bal build --graalvm
+```
+
+The `--graalvm` flag triggers the GraalVM native image builder, which compiles the Ballerina code and its dependencies into a single, self-contained executable, which contains the modules in the current package, their dependencies, Ballerina runtime, and statically linked native code from the JDK.
+
+For more information on the necessary steps and additional considerations to ensure a successful compilation process of building a native executable with Ballerina and GraalVM, see [Build a GraalVM executable](https://ballerina.io/learn/graalvm-executable-overview/). 
+
+## Aggregation and grouping
+
+The language now supports the `group by` and `collect` clauses to perform aggregation and grouping-related operations as shown in the example below. 
+
+### Aggregation
+
+The `collect` clause categorizes a collection into one group as shown in the example below.
+
+```ballerina
+import ballerina/io;
+
+type Order record {|
+    int orderId;
+    string itemName;
+    float price;
+    int quantity;
+|};
+
+public function main() returns error? {
+    Order[] orders = [
+        {orderId: 1, itemName: "Rich Dad Poor Dad", price: 23.4, quantity: 2},
+        {orderId: 1, itemName: "Rich Dad Poor Dad", price: 20.4, quantity: 1},
+        {orderId: 2, itemName: "Becoming", price: 21.5, quantity: 3},
+        {orderId: 1, itemName: "Becoming", price: 21.5, quantity: 3}
+    ];
+
+    float income = from var {price, quantity} in orders
+        let var totPrice = price * quantity
```

Comment:
Can we avoid var here? @KavinduZoysa Check other places too.

Reply:
Same in other examples too, as it will show how simple the statement is with less number of lines.

Reply:
Fixed.